app/routes/circulars.edit.$circularId/CircularEditForm.tsx

```diff
@@ -150,118 +168,180 @@ export function CircularEditForm({
+        <Grid>
+          <Grid row>
+            {circularId !== undefined && (
+              <>
+                <input type="hidden" name="circularId" value={circularId} />
+                <InputGroup className="maxw-full">
+                  <InputPrefix className="wide-input-prefix">From</InputPrefix>
+                  <TextInput
+                    autoFocus
+                    className="maxw-full"
+                    name="submitter"
+                    id="submitter"
+                    type="text"
+                    defaultValue={defaultSubmitter}
+                    onChange={(event) =>
+                      setUpdatedSubmitter(event.target.value)
+                    }
+                    required
+                    disabled={!userIsModerator}
+                  />
+                </InputGroup>
+              </>
+            )}
+          </Grid>
+          <Grid row>
             <InputGroup className="border-0 maxw-full">
-              <InputPrefix className="wide-input-prefix">From</InputPrefix>
-              <span className="padding-1">{submitter}</span>
+              <InputPrefix className="wide-input-prefix">
+                {circularId === undefined ? 'From' : 'Editor'}
+              </InputPrefix>
+              <span className="padding-1">{formattedContributor} </span>
+              <Link
+                to="/user"
+                title="Adjust how your name and affiliation appear in new GCN Circulars"
+              >
+                <Button unstyled type="button">
+                  <Icon.Edit role="presentation" /> Edit
+                </Button>
+              </Link>
             </InputGroup>
-          </>
-        )}
-        <InputGroup className="border-0 maxw-full">
-          <InputPrefix className="wide-input-prefix">
-            {circularId === undefined ? 'From' : 'Editor'}
-          </InputPrefix>
-          <span className="padding-1">{formattedContributor} </span>
-          <Link
-            to="/user"
-            title="Adjust how your name and affiliation appear in new GCN Circulars"
+          </Grid>
+          {circularId !== undefined && (
+            <Grid row>
+              <InputGroup className="border-0">
+                <InputPrefix className="wide-input-prefix">Date</InputPrefix>
+                <DatePicker
+                  defaultValue={defaultCreatedOnDate}
+                  className={classnames(styles.DatePicker, 'border-1px ')}
+                  onChange={(value) => {
+                    setDate(value ?? '')
+                  }}
+                  name="createdOnDate"
+                  id="createdOnDate"
+                  dateFormat="YYYY-MM-DD"
+                />
+              </InputGroup>
+              <InputGroup className="tablet:grid-col-auto">
+                <input
+                  type="hidden"
+                  id="createdOnTime"
+                  name="createdOnTime"
+                  value={time}
+                />
```

Comment:
Why do we still need this hidden field?

Reply:
Its an easy work around for now since the TimePicker component does not post values correctly.

 I think the issue is that the underlying ComboBox component assigns the `name`  attribute to the `select` element, but the `value` to the `input` element, so when you post it just keeps posting the default value. These are the elements:

```jsx
<select
        {...selectProps}
        className="usa-select usa-sr-only usa-combo-box__select"
        name={name}
        aria-hidden
        tabIndex={-1}
        defaultValue={state.selectedOption?.value}
        data-testid="combo-box-select">
        {options.map((option) => (
          <option key={option.value} value={option.value}>
            {option.label}
          </option>
        ))}
</select>
<Input
        {...inputProps}
        role="combobox"
        onChange={(e): void => {
          if (inputProps?.onChange) {
            // Allow a custom input onChange handler
            inputProps?.onChange(e)
          }

          dispatch({ type: ActionTypes.UPDATE_FILTER, value: e.target.value })
        }}
        onClick={(): void => dispatch({ type: ActionTypes.OPEN_LIST })}
        onBlur={handleInputBlur}
        onKeyDown={handleInputKeyDown}
        value={state.inputValue}
        focused={state.focusMode === FocusMode.Input}
        aria-owns={listID}
        aria-controls={listID}
        aria-autocomplete="list"
        aria-describedby={assistiveHintID}
        aria-expanded={state.isOpen}
        aria-activedescendant={(state.isOpen && focusedItemId) || ''}
        id={id}
        disabled={isDisabled}
/>
```

Reply:
The select value _is_ included in the form data, but typing a valid value into the box does not necessarily update the select value. In other words, the time picker has useful form data if you choose a value from the combo box, but not if you type in a valid value.

This is clearly a bug in react-uswds. Would you please submit an issue upstream and add a FIXME comment in our code here linking to it and explaining why we have the hidden input?

Reply:
I think they are already aware: https://github.com/trussworks/react-uswds/issues/2806
I can make a PR for it once I finish up this stuff

Reply:
Oh, cool! Thanks! May I suggest just adding a bit more context to the FIXME comment? Something to the effect of:

> The TimePicker component does not by itself contribute useful form data because only the <select> element has a name, and the <input> field does not. So the form data is only populated correctly if the user selects an option from the dropdown, but not if they type a valid value into the combo box.

Reply:
Yup! I will just use your wording for it